### DISCUSSION TITLE
Implement metric collectors for pgexporter

### DISCUSSION THREAD

**Discussion by nikitabugrovsky** [pgexporter/pgexporter#33] [Ideas]
It would be nice to be able to switch on/off metrics groups exposed based on the config file keys or even the command line flags.

Good examples of the functionality would be:
node_exporter collectors https://github.com/prometheus/node_exporter#collectors
mysqld_exporter collectors https://github.com/prometheus/mysqld_exporter#filtering-enabled-collectors

Implementation should include:

1. Ability to switch metric groups on/off via the pgexporter.conf. For e.g. `collectors = pg_stat_statements,pg_stat_database` which is basically a `collectors` list
2. Defaults for the situation when `collectors` key is not set in the pgexporter.conf
3. Ability to explicitly exclude certain groups of metrics. For e.g. `collectors = !pg_stat_statement,pg_stat_database` 
4. Ability disable all collectors altogether via `collectors = !all`

**Comment by jesperpedersen** [comment]
If `collectors` isn't set we should enable all queries (e.g. `collectors = all` is default). I would make `none` an alias for `!all`.

I think you can assume a finite list length in `pgexporter.h` - say 64.

Otherwise looks good !
